system/jlib/jlog.hpp

```diff
@@ -1342,4 +1348,301 @@ interface IComponentLogFileCreator : extends IInterface
+    }
+
+    void copyLogFieldNames(const StringArray * fields)
+    {
+        if (*fields)
+        {
+            ForEachItemIn(fieldsindex,*fields)
+            {
+                logFieldNames.append(fields->item(fieldsindex));
+            }
+        }
+    }
+    inline const StringArray &queryLogFieldNames() { return logFieldNames; }
+};
+
+interface IRemoteLogAccess : extends IInterface
+{
+    virtual bool fetchLog(LogAccessConditions & options, StringBuffer & returnbuf) = 0;
+
+    virtual const char * getRemoteLogAccessType() = 0;
+    virtual IPropertyTree * fetchLogMap() = 0;
+    virtual const char * fetchConnectionStr() = 0;
+};
+
+inline bool fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    LogAccessConditions logFetchoptions;
+    logFetchoptions.timeRange = timeRange;
+    logFetchoptions.filter = filter;
+    logFetchoptions.copyLogFieldNames(&cols); //ensure these fields are declared in m_logMapping->queryProp("WorkUnits/@contentcolumn")? or in LogMap/Fields?"
+
+    return logAccess.fetchLog(logFetchoptions, returnbuf);
+}
+
+inline bool fetchJobIDLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, const char *jobid, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getWUIDLogAccessFilter(jobid), timeRange, cols);
+}
+
+inline bool fetchComponentLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, const char * component, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getComponentLogAccessFilter(component), timeRange, cols);
+}
+
+inline bool fetchLogByAudience(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, MessageAudience audience, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getAudienceLogAccessFilter(audience), timeRange, cols);
+}
+
+inline bool fetchLogByClass(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, LogMsgClass logclass, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getClassLogAccessFilter(logclass), timeRange, cols);
+}
+
+//logAccessPluginConfig expected to contain connectivity and log mapping information
+typedef IRemoteLogAccess * (*newLogAccessPluginMethod_t_)(IPropertyTree & logAccessPluginConfig);
+
+class jlib_decl RemoteLogAccessLoader
+{
+public:
+    RemoteLogAccessLoader() = default;
+    ~RemoteLogAccessLoader();
+
+    template <class RemoteLogAccessPlugin>
+    static RemoteLogAccessPlugin * queryRemoteLogAccessor()
+    {
+        Owned<IPropertyTree> logAccessPluginConfig = getGlobalConfig()->getPropTree("logaccess");
+        if (!logAccessPluginConfig)
+        {
+            const char *simulatedGlobalYaml = R"!!(global:
+              logaccess:
+                name: localES
+                type: elasticstack
+                libname: libelasticstacklogaccess.so
```

Comment:
as discussed, I think this should an internal implementation detail only, not exposed to user configuration.
If need to be (I'd say at a later stage) it could be named logging-access plugins, where the impl. registered a name, e.g. "elasticstack" and the @type looked up and factoried the correct one, but I think I'd just map them directly to start with.

Reply:
done